src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala

```diff
@@ -59,75 +61,68 @@ trait RawlsBillingProjectComponent {
-      val name = rawlsProjectName.value
-      uniqueResult[RawlsBillingProjectRecord](findBillingProjectByName(name)).flatMap {
+      val projectName = rawlsProjectName.value
+      uniqueResult[RawlsBillingProjectRecord](findBillingProjectByName(projectName)).flatMap {
         case None => DBIO.successful(None)
         case Some(projectRec) =>
-          for {
-            users <- findUsersByProjectName(name).result
-          } yield Option(unmarshalBillingProject(projectRec, users.toSet))
+          findBillingProjectGroups(projectName).result.flatMap { groups =>
+            DBIO.sequence(groups.map { group =>
+              for {
+                loadedGroup <- rawlsGroupQuery.load(RawlsGroupRef(RawlsGroupName(group.groupName)))
+              } yield (ProjectRoles.withName(group.role), loadedGroup.get)
+            }) map { groups =>
+              Option(unmarshalBillingProject(projectRec, groups.toMap))
+            }
+          }
       }
     }
 
     def loadProjectUsersWithEmail(rawlsProjectName: RawlsBillingProjectName): ReadAction[Seq[RawlsBillingProjectMember]] = {
-      val name = rawlsProjectName.value
-
-      val query = for {
-        projectUser <- projectUsersQuery if projectUser.projectName === name
-        user <- rawlsUserQuery if user.userSubjectId === projectUser.userSubjectId
-      } yield (user.userEmail, projectUser.role)
+      val ownerUsersQuery = findProjectUsersWithRole(rawlsProjectName, ProjectRoles.Owner)
+      val userUsersQuery = findProjectUsersWithRole(rawlsProjectName, ProjectRoles.User)
 
-      query.result.map(_.map { case (email, role) =>
-        RawlsBillingProjectMember(RawlsUserEmail(email), ProjectRoles.withName(role))
-      })
-    }
-
-    def delete(billingProject: RawlsBillingProject): ReadWriteAction[Boolean] = {
-      val name = billingProject.projectName.value
-      val projectQuery = findBillingProjectByName(name)
+      val ownerSubgroupsQuery = findProjectSubgroupsWithRole(rawlsProjectName, ProjectRoles.Owner)
+      val userSubgroupsQuery = findProjectSubgroupsWithRole(rawlsProjectName, ProjectRoles.User)
 
-      uniqueResult[RawlsBillingProjectRecord](projectQuery).flatMap {
-        case None => DBIO.successful(false)
-        case Some(projectRec) =>
-          findUsersByProjectName(name).delete andThen projectQuery.delete map { count => count > 0 }
-      }
-    }
+      val owners = ownerUsersQuery union ownerSubgroupsQuery
+      val users = userUsersQuery union userSubgroupsQuery
 
-    def addUserToProject(userRef: RawlsUserRef, billingProjectName: RawlsBillingProjectName, role: ProjectRole): WriteAction[ProjectUsersRecord] = {
-      val record = marshalProjectUsers(userRef, billingProjectName, role)
-      projectUsersQuery insertOrUpdate record map { _ => record }
+      (owners union users).result.map(_.map{ case(email, role) => RawlsBillingProjectMember(RawlsUserEmail(email), ProjectRoles.withName(role))})
```

Comment:
I think you can do this in 2 queries instead of 4, 1 for all users and 1 for all subgroups